app/frontend/src/store/modules/auth.js

```diff
@@ -34,23 +41,46 @@ export default {
       }
       return false; // There are roles to check, but nothing in token to check against
     },
-    identityProvider: () => Vue.prototype.$keycloak.tokenParsed.identity_provider,
+    identityProvider: () =>
+      Vue.prototype.$keycloak.tokenParsed.identity_provider,
     isAdmin: (_state, getters) => getters.hasResourceRoles('chefs', ['admin']),
     isUser: (_state, getters) => getters.hasResourceRoles('chefs', ['user']),
     keycloakReady: () => Vue.prototype.$keycloak.ready,
     keycloakSubject: () => Vue.prototype.$keycloak.subject,
     moduleLoaded: () => !!Vue.prototype.$keycloak,
     realmAccess: () => Vue.prototype.$keycloak.tokenParsed.realm_access,
-    redirectUri: state => state.redirectUri,
+    redirectUri: (state) => state.redirectUri,
     resourceAccess: () => Vue.prototype.$keycloak.tokenParsed.resource_access,
     token: () => Vue.prototype.$keycloak.token,
     tokenParsed: () => Vue.prototype.$keycloak.tokenParsed,
-    userName: () => Vue.prototype.$keycloak.userName
+    userName: () => Vue.prototype.$keycloak.userName,
+
+    user: (_state, getters) => {
+      const username = getters.authenticated
+        ? getters.tokenParsed.identity_provider_identity
+          ? getters.tokenParsed.identity_provider_identity
+          : getters.tokenParsed.preferred_username
+        : '';
```

Comment:
I would recommend avoiding nested ternaries. While they definitely work the way its written, they are cognitively harder to process. It would be better to write this as a `let username = ''` and then if certain conditions are met, reassigned instead in a more classic if/else format.

This also needs to be unit tested. Refer to the `auth.getters.spec.js` file.

Reply:
Not sure we need the user object and the token object?
If there's a formIO instruction concept of `user` as well as `token` it would be helpful, as people might refer to formIO documentation. So if that case I'd leave it in (then we'd want to ask ourself if `token` should be the parsed one or the encoded one...) I haven't dug into the FormIO expected behaviour so we could check up on that and try to match it.

Reply:
I see this in the existing form designer UI, so decoded token is expected yeah

![image](https://user-images.githubusercontent.com/17445138/144684511-cd800fe6-797f-462d-baac-d16115392aef.png)

Reply:
Formio also has a user field looking into localstorage, which is why I overwrote it as well.

Reply:
https://github.com/formio/formio.js/blob/73bb21a6d158db2d6eb33250ceff2997390ac398/src/Element.js#L549

Reply:
Yeah makes sense to me if it's a formio existing field that people could come across to override that as well.